src/_P042_Candle.ino

```diff
@@ -76,422 +83,440 @@ enum ColorType {
-        }
-        Candle_bright = PCONFIG(3);
-        Candle_type = (SimType)PCONFIG(4);
-        Candle_color = (ColorType)PCONFIG(5);
+        // char tmpString[128];
+        // sprintf_P(tmpString, PSTR("<input type='text' name='brText' id='brText' size='3' value='%u'>"), PCONFIG(3));
+        // addHtml(concat(F("<input type='text' name='brText' id='brText' size='3' value='"),) + F("'>"));
+        addFormNumericBox(F("Brightness Value"), F("brText"), PCONFIG(3), 0, 255);
+      }
 
-        success = true;
-        break;
+      // Some Javascript we need to update the items
+      addHtml(F("<script script type='text/javascript'>"));
+
+      // "function update(picker) {"
+      // "    document.getElementById('wRed').value = Math.round(picker.rgb[0]);"
+      // "    document.getElementById('wGreen').value = Math.round(picker.rgb[1]);"
+      // "    document.getElementById('wBlue').value = Math.round(picker.rgb[2]);"
+      // '}'
+      // Minified:
+      addHtml(F(
+                "function update(e){document.getElementById('wRed').value=Math.round(e.channel('R')),document.getElementById('wGreen').value=Math.round(e.channel('G')),document.getElementById('wBlue').value=Math.round(e.channel('B'))}"));
+
+      // "window.addEventListener('load', function(){"
+      // "var slider = document.getElementById('brSlide');"
+      // "slider.addEventListener('change', function(){"
+      // "document.getElementById('brText').value = this.value;"
+      // "});"
+      // "});</script>"
+      // Minified:
+      addHtml(F(
+                "window.addEventListener('load',function(){document.getElementById('brSlide').addEventListener('input',function(){document.getElementById('brText').value=this.value})});"));
+      addHtml(F("</script>"));
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      PCONFIG(0) = getFormItemInt(F("wRed"));
+      PCONFIG(1) = getFormItemInt(F("wGreen"));
+      PCONFIG(2) = getFormItemInt(F("wBlue"));
+      PCONFIG(3) = getFormItemInt(F("brText"));
+      PCONFIG(4) = getFormItemInt(F("cType"));
+      PCONFIG(5) = getFormItemInt(F("clrType"));
+
+      Candle_red   = PCONFIG(0);
+      Candle_green = PCONFIG(1);
+      Candle_blue  = PCONFIG(2);
+
+      if (Candle_bright > 255) {
+        Candle_bright = 255;
       }
+      Candle_bright = PCONFIG(3);
+      Candle_type   = (SimType)PCONFIG(4);
+      Candle_color  = (ColorType)PCONFIG(5);
+
+      success = true;
+      break;
+    }
 
     case PLUGIN_INIT:
+    {
+      Candle_red    = PCONFIG(0);
+      Candle_green  = PCONFIG(1);
+      Candle_blue   = PCONFIG(2);
+      Candle_bright = PCONFIG(3);
+
+      if ((Candle_red == 0) && (Candle_green == 0) && (Candle_blue == 0)) {
+        Candle_bright = BRIGHT_START;
+      }
+      Candle_type  = static_cast<SimType>(PCONFIG(4));
+      Candle_color = static_cast<ColorType>(PCONFIG(5));
+
+      if (!Candle_pixels || (GPIO_Set == false))
```

Comment:
There's quite a bit of code overlap between `PLUGIN_INIT` and `PLUGIN_SAVE`.
The main differences are probably worth looking at to see if those actually need to be different or are actually a bug.

Reply:
Removed from PLUGIN_SAVE, as the SAVE causes an EXIT and INIT sequence...